org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/TelemetryManager.java

```diff
@@ -106,9 +114,36 @@ public void onBuildFinished(long buildFinishedTime) {
 						sourceLevelMax = Float.parseFloat(sourceLevel);
 					}
 				}
+
+				try {
+					IMarker[] projectMarkers = project.findMarkers(null /* all markers */, true /* subtypes */, IResource.DEPTH_ZERO);
+					projectErrors += Stream.of(projectMarkers).filter(m -> m.getAttribute(IMarker.SEVERITY, 0) == IMarker.SEVERITY_ERROR).count();
+					IMarker[] allmarkers = project.findMarkers(null /* all markers */, true /* subtypes */, IResource.DEPTH_INFINITE);
+					unresolvedImportErrors += Stream.of(allmarkers).filter(m -> isUnresolvedImportError(m)).count();
+				} catch (CoreException e) {
+					// ignore
+				}
+
+				IJavaProject javaProject = ProjectUtils.getJavaProject(project);
+				if (javaProject != null) {
+					try {
+						IVMInstall vmInstall = JavaRuntime.getVMInstall(javaProject);
+						if (JavaRuntime.compareJavaVersions(vmInstall, sourceLevel) < 0) {
+							jdkMismatch = true;
+						}
+					} catch (CoreException e) {
+						// ignore
+					}
+					
+				}
 			}
 		}
 
+		properties.addProperty("project.projectErrorCount", Long.toString(projectErrors));
+		properties.addProperty("project.unresolvedImportErrorCount", Long.toString(unresolvedImportErrors));
+		properties.addProperty("project.autobuild", Boolean.toString(prefs.getPreferences().isAutobuildEnabled()));
```

Comment:
Would you consider getting the settings directly from the client side ? For example, at https://github.com/redhat-developer/vscode-java/blob/master/src/telemetry.ts#L64 .

Reply:
I store the autobuild flag here to make it easier to correlate with error metrics. For example, the project error and unresolved import error rate only matter when autobuild is enabled. When autobuild is off, these errors may not be reported.